app/assets/javascripts/govuk_publishing_components/components/accordion.js

```diff
@@ -39,6 +36,10 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
     this.$module.actions.showAllText = this.$module.getAttribute('data-show-all-text')
     this.$module.actions.hideAllText = this.$module.getAttribute('data-hide-all-text')
     this.$module.actions.thisSectionVisuallyHidden = this.$module.getAttribute('data-this-section-visually-hidden')
+  }
```

Comment:
A lot of things get initialised here before the `init` function. Is that bad, or is that okay? Genuinely uncertain.

Reply:
My view is that the constructor should contain all static attributes (hence the queries and getters moved here) then the `init` should handle behavior (event listeners and/or subsequent function calls).

Reply:
And presumably there's no situation where the constructor gets called but the `init` doesn't?

Reply:
With the GOV.UK modules system there isn't. But when not used with the modules system (for example if we want to move the code to `govuk-frontend`) it gives the flexibility of initialising the script at any point in time/with any mechanishm and without querying the attributes again.

Reply:
Great, thanks for the explanation.